test_reduce_cache_miss_sub_core_grids: atol=1e-06 is physically impossible for bfloat16 (intermittent failure)

## Summary

`tests/ttnn/unit_tests/operations/reduce/test_reduction_program_cache.py::test_reduce_cache_miss_sub_core_grids` uses `atol=1e-06, rtol=1e-06` for a bfloat16 reduction, which is physically impossible to satisfy and causes intermittent CI failures.

## Details

- **Op:** `ttnn.sum(dim=-1, keepdim=True)` over 64 elements
- **Shape:** `[1, 1, 64, 64]`, **dtype:** `bfloat16`
- **Tolerance used:** `atol=1e-06, rtol=1e-06`

## Why the tolerance is wrong

For output values in the range ~33–43 (sum of 64 bfloat16 values near 0.6), one bfloat16 ULP is approximately **0.25**. A single rounding error in the reduction is therefore ~0.25 — four orders of magnitude larger than `atol=1e-06`.

The test passes or fails depending on the random seed. With different `torch.rand` draws, the accumulated rounding error either stays under 0.25 or does not. This is non-deterministic and unrelated to any real correctness issue.

## Evidence

Observed failure in CI run [24810585538/job/72615683028](https://github.com/tenstorrent/tt-metal/actions/runs/24810585538/job/72615683028):
```
AssertionError: Max ATOL Delta: 0.25, Max RTOL Delta: 0.0067138671875
atol used: 1e-06
rtol used: 1e-06
```

The same test passed on serial retry with a different random seed.

## Existing precedent

`test_reduce_cache_miss_different_input_dtypes` in the same file uses `atol=0.25, rtol=0.007, pcc_threshold=0.999` for bfloat16 — appropriate tolerances for this dtype and op.

## Fix

Update `test_reduce_cache_miss_sub_core_grids` to use the same tolerances as `test_reduce_cache_miss_different_input_dtypes` for bfloat16:
- `atol=0.25`
- `rtol=0.007`
- `pcc_threshold=0.999`
- `frobenius_threshold=0.001`

The same issue likely exists on `main`.

Prompted by @nsextonTT

--- a/tests/ttnn/unit_tests/operations/reduce/test_reduction_program_cache.py
+++ b/tests/ttnn/unit_tests/operations/reduce/test_reduction_program_cache.py
@@ -273,19 +273,19 @@ def test_reduce_cache_miss_sub_core_grids(device, isolate_program_cache):
     assert_numeric_metrics(
         torch_ref,
         ttnn.to_torch(tt_out1),
-        pcc_threshold=0.9999,
-        rtol=1e-06,
-        atol=1e-06,
-        frobenius_threshold=1e-09,
+        pcc_threshold=0.999,
+        rtol=0.007,
+        atol=0.25,
+        frobenius_threshold=0.001,
     )
     # test for equivalance
     assert_numeric_metrics(
         torch_ref,
         ttnn.to_torch(tt_out2),
-        pcc_threshold=0.9999,
-        rtol=1e-06,
-        atol=1e-06,
-        frobenius_threshold=1e-09,
+        pcc_threshold=0.999,
+        rtol=0.007,
+        atol=0.25,
+        frobenius_threshold=0.001,
     )
 
     assert device.cache_entries_counter.total == 2

--- a/tests/ttnn/unit_tests/operations/reduce/test_reduction_program_cache.py
+++ b/tests/ttnn/unit_tests/operations/reduce/test_reduction_program_cache.py
@@ -278,7 +278,7 @@ def test_reduce_cache_miss_sub_core_grids(device, isolate_program_cache):
         atol=0.25,
         frobenius_threshold=0.001,
     )
-    # test for equivalance
+    # test for equivalence
     assert_numeric_metrics(
         torch_ref,
         ttnn.to_torch(tt_out2),

